ci/scripts/rust_example.sh

```diff
@@ -29,5 +29,6 @@ do
   # Skip tests that rely on external storage and flight
   if [ ! -d $filename ]; then
      cargo run --example $example_name
+     cargo clean -p datafusion-examples
```

Comment:
I can into consistent build issues where the runner would run out of disk space. This is the best solution I could come up with. It seems to add about 5-10 seconds to the example action, which takes about 12 minutes overall.

Certainly open to alternatives.

Reply:
I am hitting similar issues with trying to add new examples in https://github.com/apache/datafusion/pull/11089

This seems like a good idea to me

My best hope is to move some of the example binaries into inline examples in the docs instead: https://github.com/apache/datafusion/pull/11178

Hopefully that will free up some additional space as well as make the examples easier to navigate

Reply:
er, sorry this came up. There are a few other tricks here (https://github.com/actions/runner-images/issues/2840#issuecomment-790492173) to clear up some disk space that may help again.

Reply:
No worries! I think this is a good excuse / reason to take another pass through the examples directory (and the library guide that you started however long ago).